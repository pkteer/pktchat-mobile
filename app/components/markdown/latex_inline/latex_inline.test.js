// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';

import LatexInline from './latex_inline';

describe('LatexCodeBlock', () => {
    const baseProps = {
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <LatexInline
                {...baseProps}
                content={'\\frac{1}{2} == 0.5^{0*\\pi} - 0.5'}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should have maxWidth 10', () => {
        const wrapper = shallow(
            <LatexInline
                {...baseProps}
                content={'\\frac{1}{2} == 0.5^{0*\\pi} - 0.5'}
                maxMathWidth={10}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
