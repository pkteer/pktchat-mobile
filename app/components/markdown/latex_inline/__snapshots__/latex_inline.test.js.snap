// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatexCodeBlock should have maxWidth 10 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": -7,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <ForwardRef
    color="black"
    math="\\\\frac{1}{2} == 0.5^{0*\\\\pi} - 0.5"
    onError={[Function]}
    renderError={[Function]}
    resizeMode="contain"
    style={
      Array [
        Object {
          "marginBottom": -7,
        },
        Object {
          "maxWidth": 10,
        },
      ]
    }
  />
</View>
`;

exports[`LatexCodeBlock should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": -7,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <ForwardRef
    color="black"
    math="\\\\frac{1}{2} == 0.5^{0*\\\\pi} - 0.5"
    onError={[Function]}
    renderError={[Function]}
    resizeMode="contain"
    style={
      Array [
        Object {
          "marginBottom": -7,
        },
        Object {
          "maxWidth": "100%",
        },
      ]
    }
  />
</View>
`;
